serde/src/test/kotlin/com/amazon/ionhiveserde/configuration/CaseInsensitiveDecoratorTest.kt

```diff
@@ -130,14 +131,33 @@ class CaseInsensitiveDecoratorTest {
 
     @Test
     fun ionStructCaseInsensitiveDecoratorCloneAndRetain() {
-        val struct = struct_for(" { Foo: 'bar' }")
+        val struct = struct_for("{ Foo: 'bar' }")
         val s = struct.cloneAndRetain("Foo")
 
         assertEquals(s.size(), 1)
         assertEquals(s.containsKey("foo"), true)
         assertStructWrapper(s)
     }
 
+    @Test
+    fun ionStructCaseInsensitiveDecoratorIteratorNext() {
+        val struct = struct_for(" { Foo: 1, Bar: '2'}")
+        val iter = struct.iterator()
+        val foo = iter.next()
+        assertEquals(foo, ION.newInt(1))
+        val bar = iter.next()
+        assertEquals(bar, ION.newSymbol("2"))
+    }
+
+    @Test
+    fun ionStructCaseInsensitiveDecoratorIteratorNull() {
+        val struct = struct_for(" { Foo: null }")
+        val iter = struct.iterator()
+        val v = iter.next()
+        // Returns Ion null like what Ion container does
+        assertEquals(v, ION.newNull())
```

Comment:
CaseInsensitiveDecorator's iterator should behavior the same as regular Ion container iterator. It should returns Ion null instead of null.